desci-server/src/controllers/auth/issueApiKey.ts

```diff
@@ -0,0 +1,74 @@
+import { Request, Response } from 'express';
+
+import { prisma } from '../../client.js';
+import { logger as parentLogger } from '../../logger.js';
+import { magicLinkRedeem } from '../../services/auth.js';
+
+import { generateApiKey, hashApiKey } from './utils.js';
+
+interface IssueApiKeyResponse {
+  ok: boolean;
+  apiKey?: string;
+  error?: string;
+}
+
+export async function issueApiKey(req: Request, res: Response<IssueApiKeyResponse>) {
+  // debugger;
+  const logger = parentLogger.child({
+    module: 'AUTH::issueApiKeyController',
+    body: req.body,
+    user: { id: (req as any).user?.id },
+  });
+  logger.trace({}, '[API KEY] Issue API key');
+
+  try {
+    const { memo, magicToken } = req.body;
+    const user = (req as any).user;
+    const createdIp = req.ip;
+
+    if (!memo) return res.status(400).json({ ok: false, error: 'Unique Memo required' });
+    if (!magicToken) return res.status(400).json({ ok: false, error: 'Magic Token required' });
```

Comment:
Tangential, but why do we combine the `{ ok: boolean }` with error codes? I've seen this in several places. I feel like the status would be enough.